src/scrapy_folder_tree/trees/hash.py

```diff
@@ -4,10 +4,10 @@
 
 
 class HashTree(TreeBase):
-    def __init__(self, depth: int) -> None:
+    def __init__(self, depth: int, length: int) -> None:
         self.DEPTH = depth
+        self.LENGTH = length
 
     def build_path(self, filepath) -> str:
         filename = os.path.basename(filepath)
-        dir_name = os.path.dirname(filepath)
-        return os.path.join(dir_name, *[_ for _ in filename[: self.DEPTH]], filename)
+        return os.path.join(*[filename[i:i + self.LENGTH] for i in range(0, self.DEPTH * self.LENGTH, self.LENGTH)], filename)
```

Comment:
Pretty clean implementation @HemantSachdeva . Btw, `dirname` should remain as is in order to store the files in the necessary folder (regardless of the extra folder structure)
```suggestion
        dir_name = os.path.dirname(filepath)
        return os.path.join(dir_name, *[filename[i:i + self.LENGTH] for i in range(0, self.DEPTH * self.LENGTH, self.LENGTH)], filename)
```

Reply:
Sure  I'll re-add `dir_name`.
I'll also update the docs as #5.